test/es-module/test-esm-assertionless-javascript-import.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+const common = require('../common');
+const { strictEqual } = require('assert');
+
+async function test() {
+  {
+    const [secret0, secret1] = await Promise.all([
+      import('../fixtures/es-modules/default-export.mjs'),
+      import(
+        '../fixtures/es-modules/default-export.mjs',
+        { assert: { type: 'javascript' } }
+      ),
+    ]);
+
+    strictEqual(secret0.default.ofLife, 42);
+    strictEqual(secret1.default.ofLife, 42);
+    strictEqual(secret0.default, secret1.default);
+    strictEqual(secret0, secret1);
+  }
+
+  {
+    const [secret0, secret1] = await Promise.all([
+      import('../fixtures/es-modules/default-export.mjs?test'),
+      import(
+        '../fixtures/es-modules/default-export.mjs?test',
+        { assert: { type: 'javascript' } }
+      ),
+    ]);
+
+    strictEqual(secret0.default.ofLife, 42);
+    strictEqual(secret1.default.ofLife, 42);
+    strictEqual(secret0.default, secret1.default);
+    strictEqual(secret0, secret1);
+  }
+
+  {
+    const [secret0, secret1] = await Promise.all([
+      import('../fixtures/es-modules/default-export.mjs#test'),
+      import(
+        '../fixtures/es-modules/default-export.mjs#test',
+        { assert: { type: 'javascript' } }
+      ),
+    ]);
+
+    strictEqual(secret0.default.ofLife, 42);
+    strictEqual(secret1.default.ofLife, 42);
+    strictEqual(secret0.default, secret1.default);
+    strictEqual(secret0, secret1);
+  }
+
+  {
+    const [secret0, secret1] = await Promise.all([
+      import('../fixtures/es-modules/default-export.mjs?test2#test'),
+      import(
+        '../fixtures/es-modules/default-export.mjs?test2#test',
+        { assert: { type: 'javascript' } }
+      ),
+    ]);
+
+    strictEqual(secret0.default.ofLife, 42);
+    strictEqual(secret1.default.ofLife, 42);
+    strictEqual(secret0.default, secret1.default);
+    strictEqual(secret0, secret1);
+  }
+
+  {
+    const [secret0, secret1] = await Promise.all([
+      import('data:text/javascript,export default { ofLife: 42 }'),
+      import(
+        'data:text/javascript,export default { ofLife: 42 }',
+        { assert: { type: 'javascript' } }
+      ),
+    ]);
+
+    strictEqual(secret0.default.ofLife, 42);
+    strictEqual(secret1.default.ofLife, 42);
+  }
```

Comment:
Maybe add a test case where the assertionless import comes after.

Reply:
Done. I duplicated (and flipped the order of the `import` statements) of the first test rather than the last one, because the first one was more extensive.